Flutter Test (Fernanda Dias)

fernandasdias opened the pull request.

jeffs-sf commented on README.md:
```diff
@@ -2,6 +2,51 @@
+
+### IDE Setup
+
+<details>
+<summary>Use with VSCode</summary>
+<p>
```
**info:** these are collapsibles since some might already be familiar with the approach. I didn't want to push down the rest of the content too far.

jeffs-sf commented on README.md:
```diff
@@ -51,7 +96,7 @@ Feel free to add more tests as you see fit but the above is the minimum requirem
 
 - See this [Figma File](https://www.figma.com/file/UOQDbU02GG2yaJMfrO9q9d/Flutter-Test?node-id=0%3A1) for design specifics like
-fonts, themes, colors, etc.
+  fonts, themes, colors, etc.
```
**info:** auto formatted. no diff.

jeffs-sf commented on README.md:
```diff
@@ -111,7 +156,7 @@ While the purpose of this challenge is not to guage whether you can achieve 100%
 
 - Implement persistance storage for favorited restaurants
-- Implement the *Bonus Detail Screen*
+- Implement the _Bonus Detail Screen_
```
**info:** auto formatted. no diff.

brianb-sf reviewed (approved):
+1 😄

fernandasdias committed:
Building presenter layer

fernandasdias committed:
Buid presentation layer with the restaurants list view

fernandasdias committed:
Build usecases

fernandasdias committed:
Build widgets

fernandasdias committed:
build repositories

fernandasdias committed:
pubspec dependencies

fernandasdias committed:
Building the restaurant detail page

fernandasdias committed:
Working on the favorites button and state management

fernandasdias committed:
cleanup

fernandasdias committed:
Widget testing... initial thoughts

fernandasdias committed:
Create cubit_test.dart

fernandasdias committed:
dependencies updated

fernandasdias committed:
Update .gitignore

fernandasdias committed:
favorites page view

fernandasdias committed:
Update restaurant_cubit.dart

fernandasdias committed:
coverage report

fernandasdias committed:
Update .gitignore

fernandasdias committed:
iOS configs

fernandasdias committed:
clean up tests folder

fernandasdias committed:
cleanups

fernandasdias committed:
automating tests + coverage report

fernandasdias committed:
chore: improving scalability of the network and datasource layers

Added some refactorings to improve scalability, thinking about the future possibilities, trying to make reusable networking layers, separating responsibilities that previously were all inside the repository.

fernandasdias committed:
chore: treating null safe in the frontend

fernandasdias committed:
Update sample_json.dart

fernandasdias committed:
refactor: moving test files to different folders

fernandasdias committed:
Update pubspec.yaml

fernandasdias committed:
chore: adding note

fernandasdias committed:
iOS essentials

fernandasdias committed:
capitalizing letters

fernandasdias committed:
chore: updating coverage report
